man/coap_encryption.txt.in

```diff
@@ -84,20 +84,306 @@ In principle the set-up sequence for CoAP Clients looks like
 ----
 
 Multiple client sessions are supported per Context.
 
 Due to the nature of TLS, there are Callbacks that are invoked as the TLS
 session negotiates encryption algorithms, encryption keys etc.
 Where possible, the CoAP layer handles all this automatically based on
-different configuration options passed in by the coap_*_pki() functions.
+different configuration options passed in by the coap_context_set_pki(),
+coap_new_client_session_pki(), coap_context_set_psk2() and
+coap_new_client_session_psk2() functions.
+
+PSK CLIENT INFORMATION
+----------------------
+
+For Client PSK setup, the required information needs to be provided in the setup
+calls with optional application Callbacks defined to update the Identity and
+PSK. Initially, the Client has to provide an Identity and a PSK.
+
+Libcoap will put the Identity and PSK as appropriate into the TLS environment.
+
+*SECTION: PSK Client: coap_dtls_cpsk_t*
+[source, c]
+----
+typedef struct coap_dtls_cpsk_t {
+  uint8_t version; /* COAP_DTLS_CL_PSK_SETUP_VERSION */
+
+  /* Options to enable different TLS functionality in libcoap */
+  uint8_t reserved[7];             /* Reserved - must be set to 0 for
+                                      future compatibility */
+
+  /** Identity Hint check call-back function.
+   * If not @p NULL, is called when the Identity Hint (TLS1.2 or earlier) is
+   * provided by the server.
+   * The appropriate Identity and PSK to use can then be returned.
+   */
+  coap_dtls_ih_callback_t validate_ih_call_back;
+  void *ih_call_back_arg;  /* Passed in to the Identity Hint call-back
+                              function */
+
+  char* client_sni;    /*  If not NULL, SNI to use in client TLS setup.
+                           Owned by the client app and must remain valid
+                           during the call to coap_new_client_session_pki().
+                           Note: Not supported by TinyDTLS. */
+
+  coap_dtls_cpsk_key_t psk_key;  /* Client PSK key definition */
+} coap_dtls_cpsk_t;
+----
+
+More detailed explanation of the coap_dtls_cpsk_t structure follows.
+
+*WARNING*: For all the parameter definitions that are pointers to other
+locations, these locations must remain valid during the lifetime of all the
+underlying TLS sessions that are, or will get created based on this PSK
+definition.
+
+*SECTION: PSK Client: coap_dtls_cpsk_t: Version*
+[source, c]
+----
+#define COAP_DTLS_CL_PSK_SETUP_VERSION 1
+----
+
+*version* is set to COAP_DTLS_CL_PSK_SETUP_VERSION.  This will then allow
+support for different versions of the coap_dtls_cpsk_t structure in the future.
+
+*SECTION: PSK Client: coap_dtls_cpsk_t: Reserved*
+
+*reserved* All must be set to 0.  Future functionality updates will make use of
+these reserved definitions.
+
+*SECTION: PSK Client: coap_dtls_cpsk_t: Identity Hint Callback*
+[source, c]
+----
+/**
+ * Identity Hint Validation call-back that can be set up by
```

Comment:
Inconsistent spelling of "callback" (vs. "call-back")

Reply:
All call-back have been replaced with callback (including in the libcoap .h files).